addons/mail/static/src/core/common/composer_actions.js

```diff
@@ -132,6 +132,18 @@ composerActionsRegistry
         name: _t("Open Full Composer"),
         onClick: (component) => component.onClickFullComposer(),
         sequence: 30,
+    })
+    .add("add-canned-response", {
+        condition: (component) =>
+            component.store.hasCannedResponses &&
```

Comment:
Some thought about `hasCannedResponses` and the behavior of the `::` delimiter if no canned responses are created yet.

`hasCannedResponses` is only checked once when the page loads. However, canned responses can be created or deleted, resulting in an outdated state. Since the composer action aims to encourage the use of canned responses, could we display a message notifying the user that no canned responses are currently available and provide a link to the creation form?

For search features like `"#/@"`, it makes sense to hide the list if nothing matches the search terms because the list is initially displayed. However, with `"::"`, nothing happens if no canned responses are created, which effectively hides the feature.